lib/java/com/google/android/material/appbar/HeaderScrollingViewBehavior.java

```diff
@@ -72,8 +72,7 @@ public boolean onMeasureChild(
           if (ViewCompat.getFitsSystemWindows(header)) {
             final WindowInsetsCompat parentInsets = parent.getLastWindowInsets();
             if (parentInsets != null) {
-              availableHeight += parentInsets.getSystemWindowInsetTop()
-                  + parentInsets.getSystemWindowInsetBottom();
+              availableHeight += parentInsets.getSystemWindowInsetTop();
```

Comment:
Can you provide more details in the commit message about why this fixes the issue and what will be the impact if we count bottom insets into available heights? Some screenshots will be even better. Thanks!

Reply:
**Description:** The available height for the scrolling content includes the bottom inset. As a consequence, the scrolling content is laid out behind the keyboard.

**Expected behavior:** The available height for the scrolling content does not include the bottom inset. As a consequence, the scrolling content is laid out above the keyboard.